### PR TITLE
SLI-1342 Export configuration file

### PR DESCRIPTION
**Pull request by eray-felek-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java [hunk 1]
```diff
@@ -306,23 +314,59 @@ public void actionPerformed(ActionEvent e) {
...
+          filename = "connectedMode.json";
+        }
+
+        var directoryPath = root.resolve(".sonarlint");
+        var fullFilePath = directoryPath.resolve(filename);
+
+        try {
+          Files.createDirectories(directoryPath);
+        } catch (IOException e) {
+          SonarLintConsole.get(project).error("Error while creating the directory, IO exception : " + e.getMessage());
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java:349]: We should not continue is there is an error here

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java [hunk 2]
```diff
@@ -306,23 +314,59 @@ public void actionPerformed(ActionEvent e) {
...
+        }
+
+        try (var writer = new BufferedWriter(new FileWriter(String.valueOf(fullFilePath), StandardCharsets.UTF_8))) {
+          writer.write(sharedFileContent.getJsonFileContent());
```
[the remark this answers is not in the document]
Comment (eray-felek-sonarsource) [reply src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java:353]: Hi, this was discussed and the decision was we overwrite silently. The justification for the decision was that we already have a confirmation pop up with file name and content warning the users.

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java [hunk 3]
```diff
@@ -306,23 +316,71 @@ public void actionPerformed(ActionEvent e) {
...
+
+        var directoryPath = root.resolve(".sonarlint");
+        var fullFilePath = directoryPath.resolve(filename);
+
+        try {
+          Files.createDirectories(directoryPath);
+        } catch (IOException e) {
+          SonarLintProjectNotifications.Companion.get(project).simpleNotification(
+            null,
+            "Could not create the directory .sonarlint. Please check the logs for more details",
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java:353]:
```suggestion
            "Could not create the directory '.sonarlint', please check the logs for more details",
```

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java [hunk 4]
```diff
@@ -306,23 +316,71 @@ public void actionPerformed(ActionEvent e) {
...
+
+          return;
+        }
+
+        try (var writer = new BufferedWriter(new FileWriter(String.valueOf(fullFilePath), StandardCharsets.UTF_8))) {
+          writer.write(sharedFileContent.getJsonFileContent());
+        } catch (IOException e) {
+          SonarLintProjectNotifications.Companion.get(project).simpleNotification(
+            null,
+            "Could not create the file : " + filename + ". Please check the logs for more details",
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java:366]:
```suggestion
            "Could not create the file : '" + filename + "', please check the logs for more details",
```

**Review by nquinquenel — APPROVED** [review]

**Pull request merged**